_gsocproposals/2020/proposal_RucioSupportBot.md

```diff
@@ -0,0 +1,46 @@
+The Native Language Processing (NLP) is nowadays fairly developed technique used in many fields with need of analyzing large chunks of text and automate actions on them. Next to a few other use cases in Rucio project, an important one is to process responses and questions of users and to provide satisfying answers - development of a prototype of a Bot that is capable to handle user requests until certain complexity. The request would be handled by experts only in case when the bot can not provide a reliable answer. Such a complex task consists of the following milestones.
+
+## Milestones
+
+ * Input Data: 
+   * creation of corpora or use of existing one 
+   * getting emails from the support mailing lists, preferably through IMAP client
+ * Extracting meanings of the input data and acquiring the data for further analyses, preferably with the NLTK.
+   * Creation of a training set and valdiation set from the input data. 
+   * Development of a tool capable to extract the meanings.
+ * Sorting out possible solutions from provided sources (e.g. documentation, docstrings, table of answers) and triggering a response to the user.
+ * Fallback to the Rucio experts in case that no answer could be found.
+ * Analysis of the acquired data:
+   * Feedbacks from users.
+   * Data popularity.
+   * Most frequent questions/answers.
+
+## Expected results
+
+Note that the whole Rucio Bot project regards full engagement for time period longer than the GSoC timescale. A minimal success of the project would be declared in case when the first two bullets of the milestones are done and suggestions for the rest is provided. Especially the second bullet requires deep expertise with NLTK including creation of both the training set and the validation set and performing the developed bot on them. The validation set provides us with statistics necessary to demonstrate reliability of the bot. The other bullets are at least as advanced as the second one. Providing a prototype of a code capable to process the tasks from the last three bullets would be considered as advanced goals of the GSoC project. Working out the most optimal solution can again mean analyzing text from documentation or docstrings, while for easy requests, there can be a list of pre-defined answers or cached answers from the previous cases.
```

Comment:
Please rephrase, to avoid misunderstanding that it is a requirement for the summer student to engage for more than the GSoC exercise period. The milestones should also explicitly separate the part you consider essential and achievable during the summer project, and the optional part. You could even at the limit present only the essential part as "milestones" and the rest as "ideas for extension", or put the extension part in a separate doc, linked in this proposal.

Reply:
Also, "suggestions for the rest" is not really a deliverable, please rephrase.

Reply:
"considered as advanced goals of the GSoC project" - if these are optional, rather do not put them in the list of goals, GSoC practically works on "pass or reject" and not on excellence criteria. Nothing prevents you to describe in detail the whole bot project, but write as goals only what you actually consider achievable and acceptable.

Reply:
I basicaly removed the description of the advanced tasks and left it just under ideas for extension.